testing/e2e/suites/nginxIngressController.go

```diff
@@ -100,61 +98,61 @@ func nicWebhookTests(in infra.Provisioned) []test {
 					return fmt.Errorf("creating client: %w")
 				}
 
-				testNIC := manifests.NewNginxIngressController("default", existingOperatorIngressClass)
-				var copyNIC v1alpha1.NginxIngressController
-				err = c.Get(ctx, client.ObjectKeyFromObject(testNIC), &copyNIC)
-				if err != nil {
-					return fmt.Errorf("get default nic: %w", err)
-				}
-
-				copyNIC.Spec.LoadBalancerAnnotations = map[string]string{
+				privateNic := manifests.NewNginxIngressController("private", "private.ingress.class")
+				privateNic.Spec.LoadBalancerAnnotations = map[string]string{
 					"service.beta.kubernetes.io/azure-load-balancer-internal": "true",
 				}
-
-				lgr.Info("updating NIC to internal")
-				if err := c.Update(ctx, &copyNIC); err != nil {
-					return fmt.Errorf("updating NIC to internal: %w", err)
+				if err := upsert(ctx, c, privateNic); err != nil {
+					return fmt.Errorf("ensuring private NIC: %w", err)
 				}
 
-				if err := wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
-					lgr.Info("validating service is updated with new annotations")
-
-					var serviceCopy corev1.Service
-					if err := c.Get(ctx, client.ObjectKey{Namespace: "app-routing-system", Name: "nginx"}, &serviceCopy); err != nil {
-						if apierrors.IsNotFound(err) {
-							lgr.Info("nginx service not found")
-							return false, nil
-						}
-
-						return false, fmt.Errorf("getting nginx service: %w", err)
+				var service v1alpha1.ManagedObjectReference
+				lgr.Info("waiting for service associated with private NIC to be ready")
+				if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+					lgr.Info("checking if private NIC service is ready")
+					var nic v1alpha1.NginxIngressController
+					if err := c.Get(ctx, client.ObjectKeyFromObject(privateNic), &nic); err != nil {
+						return false, fmt.Errorf("get private nic: %w", err)
 					}
 
-					if _, ok := serviceCopy.ObjectMeta.Annotations["service.beta.kubernetes.io/azure-load-balancer-internal"]; !ok {
-						lgr.Info("nginx annotations not found")
+					if nic.Status.ManagedResourceRefs == nil {
 						return false, nil
 					}
 
-					return true, nil
+					for _, ref := range nic.Status.ManagedResourceRefs {
+						if ref.Kind == "Service" {
+							lgr.Info("found service")
+							service = ref
+							return true, nil
+						}
+					}
+
+					lgr.Info("service not found")
+					return false, nil
 				}); err != nil {
-					return fmt.Errorf("waiting for updated nginx service: %w", err)
+					return fmt.Errorf("waiting for private NIC to be ready: %w", err)
```

Comment:
nit: typo?

Reply:
what is the typo?

Reply:
whoops - it's the line below
lgr.Info("validating service is contains private annotations")